pkgs/registry/default.nix

```diff
@@ -0,0 +1,78 @@
+{ self, pkgs, modulesLocks }:
+with pkgs.lib;
+let
+  mapping = import ../upgrade-maps/mapping.nix pkgs;
+  isTerminal = mod: !((mapping.${mod} or false).auto or false);
+  shortModuleId = (module-id:
+    let
+      parts = strings.splitString ":" module-id;
+    in
+    elemAt parts 0
+  );
+  latestModulesLocks = filterAttrs (module-id: _: isTerminal module-id) modulesLocks;
+  latestModules = mapAttrs
+    (
+      module-id: _:
+        let
+          module = self.modules.${shortModuleId module-id};
+          module-info = (builtins.fromJSON (builtins.unsafeDiscardStringContext module.text));
+        in
+        module-info
+    )
+    (filterAttrs (module-id: _: builtins.hasAttr (shortModuleId module-id) self.modules) latestModulesLocks);
+  registry = foldlAttrs
+    (acc: moduleId: module:
+      {
+        languageServers = acc.languageServers ++
+          mapAttrsToList
+            (lsp-id: lsp: {
+              id = lsp-id;
+              inherit moduleId;
+
+            } // lsp)
+            module.languageServers;
+        formatters = acc.formatters ++
+          mapAttrsToList
+            (formatter-id: formatter: {
+              id = formatter-id;
+              inherit moduleId;
+            } // formatter)
+            module.formatters;
+        packagers = acc.packagers ++
+          mapAttrsToList
+            (packager-id: packager: {
+              id = packager-id;
+              inherit moduleId;
+            } // packager)
+            module.packagers;
+        modules = acc.modules ++ [
+          {
+            id = moduleId;
+            inherit (module) name description displayVersion;
+          }
+        ];
+      }
+    )
+    {
+      languageServers = [ ];
+      formatters = [ ];
+      packagers = [ ];
+      modules = [ ];
+    }
+    latestModules;
+in
+pkgs.stdenv.mkDerivation {
+  pname = "registry";
+  version = "1";
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+  installPhase = ''
+    echo '${builtins.toJSON registry}' > $out
+  '';
+  passthru = {
+    # this allows you to query for this info wo building it:
+    # nix eval .#tools-registry.info --json
```

Comment:
```suggestion
    # nix eval .#registry.info --json
```

Reply:
also,
<img width="399" alt="image" src="https://github.com/replit/nixmodules/assets/278900/4e134a03-29ff-4bdf-b53e-9c6be9a13cb5">
shows up under `tools`, it seems to be a conflation to have build modules as well as LSPs in the tools section